tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/TestScenarioRunner.java

```diff
@@ -0,0 +1,169 @@
+package org.hawkular.apm.tests;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hawkular.apm.tests.model.TestCase;
+import org.hawkular.apm.tests.model.TestScenario;
+
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.exceptions.DockerCertificateException;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerCreation;
+import com.spotify.docker.client.messages.HostConfig;
+
+/**
+ * @author Pavol Loffay
+ */
+public class TestScenarioRunner {
+    private static final String DOCKER_MONT_DIR = "/opt/hawkular-apm-agent";
+
+    private DockerClient dockerClient;
+    private String apmVersion;
+    private HostConfig hostConfig;
+
+
```

Comment:
Not sure if you planned on doing them later, but should include method comment blocks with parameter descriptions where appropriate. Although I don't usually do these for junit tests, as this is a test harness, I think it should be properly documented to enable useful javadocs to be created.